react-sdk/src/util/registry.ts

```diff
@@ -125,32 +130,50 @@ export const mapTamboToolToContextTool = (
 /**
  * Type guard to check if a schema is a Zod 4 function schema.
  * Zod 4 function schemas have `def.type === "function"`.
+ * Uses the shared Zod schema guard from `schema.ts` to ensure the
+ * input looks like a Zod schema before inspecting internals.
  * @returns True if the schema is a Zod 4 function schema
  */
 export function isZod4FunctionSchema(schema: unknown): boolean {
-  const s = schema as Record<string, unknown>;
-  return (s.def as { type?: string })?.type === "function";
+  if (!isZodSchema(schema)) {
+    return false;
+  }
+
+  const s = schema as { def?: { type?: string } };
+  return s.def?.type === "function";
 }
 
 /**
  * Type guard to check if a schema is a Zod 3 function schema.
  * Zod 3 function schemas have `_def.typeName === "ZodFunction"`.
+ * Uses the shared Zod schema guard from `schema.ts` to ensure the
+ * input looks like a Zod schema before inspecting internals.
  * @returns True if the schema is a Zod 3 function schema
  */
 export function isZod3FunctionSchema(schema: unknown): boolean {
-  const s = schema as Record<string, unknown>;
-  return (s._def as { typeName?: string })?.typeName === "ZodFunction";
+  if (!isZodSchema(schema)) {
+    return false;
+  }
+
+  const s = schema as { _def?: { typeName?: string } };
+  return s._def?.typeName === "ZodFunction";
 }
 
 /**
  * Extracts the args schema from a Zod function schema (v3 or v4).
  * - Zod 4: args at `_def.input`
  * - Zod 3: args at `_def.args`
+ * Uses the shared Zod schema guard from `schema.ts` so we only
+ * inspect internals on values that look like Zod schemas.
  * @returns The args schema, or undefined if not found
  */
 export function getZodFunctionArgs(schema: unknown): unknown | undefined {
-  const s = schema as Record<string, unknown>;
-  const def = s._def as { input?: unknown; args?: unknown } | undefined;
+  if (!isZodSchema(schema)) {
+    return undefined;
+  }
+
+  const s = schema as { _def?: { input?: unknown; args?: unknown } };
+  const def = s._def;
   // Zod 4 uses `input`, Zod 3 uses `args`
   return def?.input ?? def?.args;
 }
```

Comment:
@CharlieHelps have this use the versioned `isZodXFunctionSchema` functions since they will also be type guards